File: examples/ruby/lib/appwrite/services/account.rb
module Appwrite
    class Account < Service

        def get()
            path = '/account'

            params = {}

            return @client.call('get', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def delete()
            path = '/account'

            params = {}

            return @client.call('delete', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def update_email(email:, password:)
            if email.nil?
                raise Appwrite::Exception.new('Missing required parameter: "email"')
            end

            if password.nil?
                raise Appwrite::Exception.new('Missing required parameter: "password"')
            end

            path = '/account/email'

            params = {}

            if !email.nil?
                params[:email] = email
            end

            if !password.nil?
                params[:password] = password
            end

            return @client.call('patch', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def get_logs()
            path = '/account/logs'

            params = {}

            return @client.call('get', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def update_name(name:)
            if name.nil?
                raise Appwrite::Exception.new('Missing required parameter: "name"')
            end

            path = '/account/name'

            params = {}

            if !name.nil?
                params[:name] = name
            end

            return @client.call('patch', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def update_password(password:, old_password: nil)
            if password.nil?
                raise Appwrite::Exception.new('Missing required parameter: "password"')
            end

            path = '/account/password'

            params = {}

            if !password.nil?
                params[:password] = password
            end

            if !old_password.nil?
                params[:oldPassword] = old_password
            end

            return @client.call('patch', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def get_prefs()
            path = '/account/prefs'

            params = {}

            return @client.call('get', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def update_prefs(prefs:)
            if prefs.nil?
                raise Appwrite::Exception.new('Missing required parameter: "prefs"')
            end

            path = '/account/prefs'

            params = {}

            if !prefs.nil?
                params[:prefs] = prefs
            end

            return @client.call('patch', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def create_recovery(email:, url:)
            if email.nil?
                raise Appwrite::Exception.new('Missing required parameter: "email"')
            end

            if url.nil?
                raise Appwrite::Exception.new('Missing required parameter: "url"')
            end

            path = '/account/recovery'

            params = {}

            if !email.nil?
                params[:email] = email
            end

            if !url.nil?
                params[:url] = url
            end

            return @client.call('post', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def update_recovery(user_id:, secret:, password:, password_again:)
            if user_id.nil?
                raise Appwrite::Exception.new('Missing required parameter: "userId"')
            end

            if secret.nil?
                raise Appwrite::Exception.new('Missing required parameter: "secret"')
            end

            if password.nil?
                raise Appwrite::Exception.new('Missing required parameter: "password"')
            end

            if password_again.nil?
                raise Appwrite::Exception.new('Missing required parameter: "passwordAgain"')
            end

            path = '/account/recovery'

            params = {}

            if !user_id.nil?
                params[:userId] = user_id
            end

            if !secret.nil?
                params[:secret] = secret
            end

            if !password.nil?
                params[:password] = password
            end

            if !password_again.nil?
                params[:passwordAgain] = password_again
            end

            return @client.call('put', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def get_sessions()
            path = '/account/sessions'

            params = {}

            return @client.call('get', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def delete_sessions()
            path = '/account/sessions'

            params = {}

            return @client.call('delete', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def get_session(session_id:)
            if session_id.nil?
                raise Appwrite::Exception.new('Missing required parameter: "sessionId"')
            end

            path = '/account/sessions/{sessionId}'
                .gsub('{sessionId}', session_id)

            params = {}

            return @client.call('get', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def delete_session(session_id:)
            if session_id.nil?
                raise Appwrite::Exception.new('Missing required parameter: "sessionId"')
            end

            path = '/account/sessions/{sessionId}'
                .gsub('{sessionId}', session_id)

            params = {}

            return @client.call('delete', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def create_verification(url:)
            if url.nil?
                raise Appwrite::Exception.new('Missing required parameter: "url"')
            end

            path = '/account/verification'

            params = {}

            if !url.nil?
                params[:url] = url
            end

            return @client.call('post', path, {
                'content-type' => 'application/json',
            }, params);
        end

        def update_verification(user_id:, secret:)
            if user_id.nil?
                raise Appwrite::Exception.new('Missing required parameter: "userId"')
            end

            if secret.nil?
                raise Appwrite::Exception.new('Missing required parameter: "secret"')
            end

            path = '/account/verification'

            params = {}

            if !user_id.nil?
                params[:userId] = user_id
            end

            if !secret.nil?
                params[:secret] = secret
            end

            return @client.call('put', path, {
                'content-type' => 'application/json',
            }, params);
        end


        protected

        private
    end 
end

File: examples/python/appwrite/services/account.py
from ..service import Service
from ..exception import AppwriteException

class Account(Service):

    def __init__(self, client):
        super(Account, self).__init__(client)

    def get(self):
        """Get Account"""

        params = {}
        path = '/account'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def delete(self):
        """Delete Account"""

        params = {}
        path = '/account'

        return self.client.call('delete', path, {
            'content-type': 'application/json',
        }, params)

    def update_email(self, email, password):
        """Update Account Email"""

        if email is None: 
            raise AppwriteException('Missing required parameter: "email"')

        if password is None: 
            raise AppwriteException('Missing required parameter: "password"')

        params = {}
        path = '/account/email'

        if email is not None: 
            params['email'] = email

        if password is not None: 
            params['password'] = password

        return self.client.call('patch', path, {
            'content-type': 'application/json',
        }, params)

    def get_logs(self):
        """Get Account Logs"""

        params = {}
        path = '/account/logs'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def update_name(self, name):
        """Update Account Name"""

        if name is None: 
            raise AppwriteException('Missing required parameter: "name"')

        params = {}
        path = '/account/name'

        if name is not None: 
            params['name'] = name

        return self.client.call('patch', path, {
            'content-type': 'application/json',
        }, params)

    def update_password(self, password, old_password = None):
        """Update Account Password"""

        if password is None: 
            raise AppwriteException('Missing required parameter: "password"')

        params = {}
        path = '/account/password'

        if password is not None: 
            params['password'] = password

        if old_password is not None: 
            params['oldPassword'] = old_password

        return self.client.call('patch', path, {
            'content-type': 'application/json',
        }, params)

    def get_prefs(self):
        """Get Account Preferences"""

        params = {}
        path = '/account/prefs'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def update_prefs(self, prefs):
        """Update Account Preferences"""

        if prefs is None: 
            raise AppwriteException('Missing required parameter: "prefs"')

        params = {}
        path = '/account/prefs'

        if prefs is not None: 
            params['prefs'] = prefs

        return self.client.call('patch', path, {
            'content-type': 'application/json',
        }, params)

    def create_recovery(self, email, url):
        """Create Password Recovery"""

        if email is None: 
            raise AppwriteException('Missing required parameter: "email"')

        if url is None: 
            raise AppwriteException('Missing required parameter: "url"')

        params = {}
        path = '/account/recovery'

        if email is not None: 
            params['email'] = email

        if url is not None: 
            params['url'] = url

        return self.client.call('post', path, {
            'content-type': 'application/json',
        }, params)

    def update_recovery(self, user_id, secret, password, password_again):
        """Create Password Recovery (confirmation)"""

        if user_id is None: 
            raise AppwriteException('Missing required parameter: "user_id"')

        if secret is None: 
            raise AppwriteException('Missing required parameter: "secret"')

        if password is None: 
            raise AppwriteException('Missing required parameter: "password"')

        if password_again is None: 
            raise AppwriteException('Missing required parameter: "password_again"')

        params = {}
        path = '/account/recovery'

        if user_id is not None: 
            params['userId'] = user_id

        if secret is not None: 
            params['secret'] = secret

        if password is not None: 
            params['password'] = password

        if password_again is not None: 
            params['passwordAgain'] = password_again

        return self.client.call('put', path, {
            'content-type': 'application/json',
        }, params)

    def get_sessions(self):
        """Get Account Sessions"""

        params = {}
        path = '/account/sessions'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def delete_sessions(self):
        """Delete All Account Sessions"""

        params = {}
        path = '/account/sessions'

        return self.client.call('delete', path, {
            'content-type': 'application/json',
        }, params)

    def get_session(self, session_id):
        """Get Session By ID"""

        if session_id is None: 
            raise AppwriteException('Missing required parameter: "session_id"')

        params = {}
        path = '/account/sessions/{sessionId}'
        path = path.replace('{sessionId}', session_id)                

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def delete_session(self, session_id):
        """Delete Account Session"""

        if session_id is None: 
            raise AppwriteException('Missing required parameter: "session_id"')

        params = {}
        path = '/account/sessions/{sessionId}'
        path = path.replace('{sessionId}', session_id)                

        return self.client.call('delete', path, {
            'content-type': 'application/json',
        }, params)

    def create_verification(self, url):
        """Create Email Verification"""

        if url is None: 
            raise AppwriteException('Missing required parameter: "url"')

        params = {}
        path = '/account/verification'

        if url is not None: 
            params['url'] = url

        return self.client.call('post', path, {
            'content-type': 'application/json',
        }, params)

    def update_verification(self, user_id, secret):
        """Create Email Verification (confirmation)"""

        if user_id is None: 
            raise AppwriteException('Missing required parameter: "user_id"')

        if secret is None: 
            raise AppwriteException('Missing required parameter: "secret"')

        params = {}
        path = '/account/verification'

        if user_id is not None: 
            params['userId'] = user_id

        if secret is not None: 
            params['secret'] = secret

        return self.client.call('put', path, {
            'content-type': 'application/json',
        }, params)
